pkg/metrics/collector/pod_ip_metrics.go

```diff
@@ -266,6 +304,68 @@ func (c *podIPMetricsCollector) fillBuckets(diff uint64) {
 	}
 }
 
+// countIPsFronRange returns the number of available hosts in a subnet.
+// The max number is limited by the size of an uint64.
+// Number of hosts is calculated with the formula:
+// IPv4: 2^x – 2, not consider network and broadcast address
+// IPv6: 2^x - 1, not consider network address
+// where x is the number of host bits in the subnet.
+func (c *podIPMetricsCollector) countIPsFromRange(subnet *net.IPNet) (uint64, error) {
+	ones, bits := subnet.Mask.Size()
+	if bits <= ones {
+		return 0, fmt.Errorf("invalid subnet mask: %v", subnet.Mask)
+	}
+	// this checks that we are not overflowing an int64
+	if bits-ones >= 64 {
+		return math.MaxUint64, nil
+	}
+	max := uint64(1) << uint(bits-ones)
+	max--
+	if subnet.IP.To4() != nil {
+		// Don't use the IPv4 network's broadcast address
+		if max == 0 {
+			return 0, fmt.Errorf("subnet includes only the network and broadcast addresses")
```

Comment:
This error message doesn't look right. The code works as:

IPv4, /32: return 0, err "subnet includes only the network and broadcast addresses"
IPv4, /31: return 0, nil
IPv4, /30: return 2, nil
IPv6, /128: return 0, nil
IPv6, /127: return 1, nil
IPv6, /126: return 3, nil

Reply:
Done.